### ISSUE TITLE
block not right

### ISSUE THREAD

**Issue by DQFSN** [Grinnode-live/grinnode.live#16]
Height：1347514

Your API return hash：0005d90115be32dd1511ef8ee7727852c0b38c59b98474612d60b46569e3d9c3

grinsan.net retrun hash: 0003565b19c043f97a79ce62de3b31345b715b6bba65088eb7f083cd1909c5fa

**Comment by stakervali** [comment]
Hello DQFSN, by return hash, did you mean last _block_pushed?

Our api returns same data with [grinmint api](https://api.grinmint.com/v2/networkStats) .

**Comment by stakervali** [comment]
When I query grinnodelive.api with that block height, it returns the correct hash starting with 000356, can you check below.

![resim](https://user-images.githubusercontent.com/60194710/129225635-54d1c3c2-88f3-410b-b0bc-afc427c88a9b.png)

**Comment by DQFSN** [comment]
![image](https://user-images.githubusercontent.com/71998518/129227843-26c0f35d-f296-46e2-ac6e-4e8f3266202b.png)

but i still get the wrong result

**Comment by stakervali** [comment]
> 
> and when i query by hash, i supposed to get no answer, bu in fact it returns the forked block.
> 


According to [api documentation](https://github.com/mimblewimble/grin-rfcs/blob/master/text/0007-node-api-v2.md#get_header) you can give either blockheight, hash as parameter. 

![resim](https://user-images.githubusercontent.com/60194710/129230167-173116eb-ce86-4824-94bb-4036a77adc10.png)

**Comment by DQFSN** [comment]
okay, I also need go to sleep, good night ~

**Comment by marekyggdrasil** [comment]
When I do

```bash
curl -d '{"jsonrpc":"2.0","id":1,"method":"get_header","params":[1347514, null, null] }' -o - https://grinnode.live/v2/foreign
```

I get

```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "result": {
    "Ok": {
      "cuckoo_solution": [
        64325325,
        75806541,
        155413856,
        173070557,
        535354060,
        827686172,
        840060193,
        861338603,
        924971382,
        938114704,
        1033442587,
        1088292991,
        1269677594,
        1341588800,
        1457533832,
        1577885553,
        1624788036,
        1810906252,
        2121963953,
        2225490864,
        2394605187,
        2441629802,
        2602419212,
        2694098538,
        2750185261,
        2768306160,
        2892345025,
        2938524678,
        3092953590,
        3109197403,
        3115062016,
        3224176701,
        3549283006,
        3571250574,
        3582217244,
        3587226909,
        3605381047,
        3722910687,
        3900129883,
        3968093287,
        4213751373,
        4225784076
      ],
      "edge_bits": 32,
      "hash": "0003565b19c043f97a79ce62de3b31345b715b6bba65088eb7f083cd1909c5fa",
      "height": 1347514,
      "kernel_mmr_size": 6003812,
      "kernel_root": "ea23183c8d21aa08d4a1c7045ab544d736318db1a7f360535d3b6ecd8dfd48a3",
      "nonce": 16401659046003505392,
      "output_mmr_size": 10166017,
      "output_root": "352ba32cea7364e62cc22bcd5cef13da7ea9633bbafbbc24b0f3560047dd4588",
      "prev_root": "ec91249ca32178fb51ceb9ad480f8bbe05afbca867a1903e80c0a452095da152",
      "previous": "0002d18d405f15dea80579e9ddc29fce14d1dd45f09ac75dabb3853fb881f5de",
      "range_proof_root": "f1e65b96987a6d82ffcc9ea69c7226885a0382c824c13e7fe0886aff48bf422d",
      "secondary_scaling": 0,
      "timestamp": "2021-08-12T13:34:22+00:00",
      "total_difficulty": 1801905797790454,
      "total_kernel_offset": "dccc3b12005306f08930ed2f2016783e7db06261a1add0938dd1924da08cb2d4",
      "version": 5
    }
  }
```

the hash it responded to me is  `0003565b19c043f97a79ce62de3b31345b715b6bba65088eb7f083cd1909c5fa` which matches the result from grinscan. Could you @DQFSN provide the exact request you are executing so that we can try again to reproduce?

**Comment by DQFSN** [comment]
curl -d '{"jsonrpc":"2.0","id":1,"method":"get_header","params":[null, "0005d90115be32dd1511ef8ee7727852c0b38c59b98474612d60b46569e3d9c3", null] }' -o - https://grinnode.live/v2/foreign


```json


{
  "id": 1,
  "jsonrpc": "2.0",
  "result": {
    "Ok": {
      "cuckoo_solution": [
        36898201,
        305384014,
        324313027,
        371221843,
        876890588,
        996277344,
        1047728405,
        1149751829,
        1204490577,
        1257136670,
        1363096407,
        1464082320,
        1559856350,
        1762481729,
        1881544234,
        1897993825,
        1918137435,
        2025761451,
        2064710345,
        2185483098,
        2269683722,
        2643488635,
        2739542765,
        2892512451,
        3083904520,
        3127744031,
        3150917364,
        3201088400,
        3347267164,
        3428832472,
        3501940253,
        3531314774,
        3579577392,
        3705753839,
        3708020612,
        3734928430,
        3821636930,
        3955051981,
        3980265647,
        4110301206,
        4125511688,
        4176933543
      ],
      "edge_bits": 32,
      "hash": "0005d90115be32dd1511ef8ee7727852c0b38c59b98474612d60b46569e3d9c3",
      "height": 1347514,
      "kernel_mmr_size": 6003812,
      "kernel_root": "b4a4e8d90fedfb84b4f6c18606698f5b1584faf952d998b63e9db4bf6ad3256e",
      "nonce": 15547289033447361717,
      "output_mmr_size": 10166017,
      "output_root": "a815112eb2a7b71e722a351b040f8fc28cca39b5fd439ee05f927daaee163558",
      "prev_root": "ec91249ca32178fb51ceb9ad480f8bbe05afbca867a1903e80c0a452095da152",
      "previous": "0002d18d405f15dea80579e9ddc29fce14d1dd45f09ac75dabb3853fb881f5de",
      "range_proof_root": "b993f232e2b3e0cc58b13c0bdc04a45a31e65689e39efdc97614569c57fe4b30",
      "secondary_scaling": 0,
      "timestamp": "2021-08-12T13:34:37+00:00",
      "total_difficulty": 1801905797790454,
      "total_kernel_offset": "dccc3b12005306f08930ed2f2016783e7db06261a1add0938dd1924da08cb2d4",
      "version": 5
    }
  }
}
```

I suppose it would return err when i query by  an orphan block's hash

**Comment by marekyggdrasil** [comment]
If we query by block height `1347514` it gives hash `0003565b19c043f97a79ce62de3b31345b715b6bba65088eb7f083cd1909c5fa` just like grinscan. If we query by hash it gives block height `1347514` just as expected.

Is the problem that if you query the hash `0005d90115be32dd1511ef8ee7727852c0b38c59b98474612d60b46569e3d9c3` you get block height `1347514`, same as for hash `0003565b19c043f97a79ce62de3b31345b715b6bba65088eb7f083cd1909c5fa`. Did I understood the problem correctly? @DQFSN

**Comment by DQFSN** [comment]
Yes, you got it

**Comment by marekyggdrasil** [comment]
Thanks for your patience and clarifying. To me it seems like an expected behaviour because it should be possible to query the orphaned blocks and they would of course have the duplicate height.

I sent message on Telegram groups and Keybase just to double check if that is an expected behaviour and I will get back to you.

**Comment by DQFSN** [comment]
Okay, thank you

**Comment by marekyggdrasil** [comment]
John comments in the Telegram ツ community group that it is very common because of one minute block time. We are both not sure how long the orphaned blocks would be stored.

If that is important to know, like if it is part of the design of your project, I would be happy to help review node source code and find out more details.

In the mean time, I recommend closing this issue as we found out the reason of block height conflict and we confirmed API is fine.
